scipy/fftpack/helper.py

```diff
@@ -215,15 +216,15 @@ def _init_nd_shape_and_axes_impl(x, shape, axes):
             raise ValueError("all axes must be unique")
 
     if not noshape:
-        shape = np.atleast_1d(shape)
+        if not isinstance(shape, Iterable):
+            shape = (shape,)
 
-        if shape.size == 0:
-            shape = shape.astype(np.intc)
+        try:
+            shape = [operator.index(s) for s in shape]
+        except TypeError as e:
+            raise ValueError("when given, shape values must be a scalar "
+                             "or iterable of integers") from e
```

Comment:
Two instances of the same thing, and we probably have others elsewhere e.g. in `scipy.signal`. Worth splitting into a little helper like `_iterable_of_int(x, name='axes')` or so?